go/worker/sentry/worker.go

```diff
@@ -0,0 +1,123 @@
+package sentry
+
+import (
+	"fmt"
+
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/grpc"
+	"github.com/oasislabs/oasis-core/go/common/identity"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/sentry"
+	"github.com/oasislabs/oasis-core/go/sentry/api"
+)
+
+const (
+	CfgEnabled     = "worker.sentry.enabled"
+	CfgControlPort = "worker.sentry.control_port"
+)
+
+// Flags has the configuration flags.
+var Flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+// Enabled returns true if Sentry worker is enabled.
+func Enabled() bool {
+	return viper.GetBool(CfgEnabled)
+}
+
+type Worker struct {
+	enabled bool
+
+	backend api.Backend
+
+	grpcServer *grpc.Server
+
+	quitCh chan struct{}
+	initCh chan struct{}
+
+	logger *logging.Logger
+}
+
+// Name returns the service name.
+func (w *Worker) Name() string {
+	return "sentry worker"
+}
+
+// Start starts the service.
+func (w *Worker) Start() error {
+	if !w.enabled {
+		w.logger.Info("not starting sentry worker as it is disabled")
+
+		return nil
+	}
+
+	// Start the sentry gRPC server.
+	if err := w.grpcServer.Start(); err != nil {
+		w.logger.Error("failed to start sentry gRPC server",
+			"err", err,
+		)
+		return err
+	}
+
+	close(w.initCh)
+
+	return nil
+}
+
+// Stop halts the service.
+func (w *Worker) Stop() {
+	if !w.enabled {
+		close(w.quitCh)
+		return
+	}
+
+	// TODO: How to close the gRPC server?
+}
+
+// Enabled returns true if worker is enabled.
+func (w *Worker) Enabled() bool {
+	return w.enabled
+}
+
+// Quit returns a channel that will be closed when the service terminates.
+func (w *Worker) Quit() <-chan struct{} {
+	return w.quitCh
+}
+
+// Cleanup performs the service specific post-termination cleanup.
+func (w *Worker) Cleanup() {
+}
+
+func New(backend api.Backend, identity *identity.Identity) (*Worker, error) {
+	w := &Worker{
+		enabled: Enabled(),
+		backend: backend,
+		quitCh:  make(chan struct{}),
+		initCh:  make(chan struct{}),
+		logger:  logging.GetLogger("worker/sentry"),
+	}
+
+	if w.enabled {
+		grpcServer, err := grpc.NewServer(&grpc.ServerConfig{
+			Name:        "sentry",
+			Port:        uint16(viper.GetInt(CfgControlPort)),
+			Certificate: identity.TLSCertificate,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("worker/sentry: failed to create a new gRPC server: %w", err)
+		}
+		w.grpcServer = grpcServer
+		// Initialize and register the sentry gRPC service.
+		sentry.NewGRPCServer(w.grpcServer.Server(), backend)
+	}
+
+	return w, nil
+}
+
+func init() {
+	Flags.Bool(CfgEnabled, false, "Enable Sentry worker")
```

Comment:
Maybe update help descriptions to make it really obvious which options are for sentry nodes and which are for sentry node consumers.

Reply:
I tried :slightly_smiling_face:. Please, take a look.

Reply:
:+1: